docs/book/src/topics/best-practices.md

```diff
@@ -0,0 +1,29 @@
+# Best Practices
```

Comment:
Should also have best practices for RBAC like not allowing users to edit `SecretProviderClassByPodStatus` CRDs

Reply:
I also like this section from the gcp provider: https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp#security-considerations section should we add them here as well?

Reply:
Added an item for this. Hopefully it provides the right context/call to action?